src/App.js

```diff
@@ -1,19 +1,49 @@
 import React from 'react';
 import './App.css';
 
+import todos from './api/todos';
 import users from './api/users';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+import { TodoList } from './components/TodoList';
+import { NewTodoForm } from './components/NewTodoForm';
+
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId),
+}));
+
+class App extends React.PureComponent {
+  state = {
+    todos: preparedTodos,
+  };
+
+  addTodo = (user, title) => {
+    const newTodo = {
+      id: Math.random() * 1000,
```

Comment:
Though it's a very low probability, there may be two unique keys if using Math.random